src/util/result.cpp

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+#include <util/result.h>
+#include <util/string.h>
+
+namespace util {
+bilingual_str ErrorDescription(const Result<void>& result)
```

Comment:
Any reason not to just include this as a method called e.g. `resullt.ErrorsStr()`?

Reply:
This is an error formatting function and I think Result class should only be responsible for returning errors, not formatting them.

Also I generally think classes with private state should have as fewer methods accessing that state to keep code readable and maintainable. If a function can use a classes public interface, it should use that instead of poking around at object internals.